### ISSUE TITLE
500 Server Error: mkdir /sys/firmware/devicetree: operation not permitted

### ISSUE THREAD

**Issue by RobertoSchneiders** [dmslabsbr/smsUps#13]
First of all, thanks for creating this add-on, it's a great idea, I have a SMS Power Sinus 3200 and I don't use any of their software but would be awesome to have some information in home assistant. 

I saw someone else had this exact issue in [this forum post ](https://forum.homeassistantbrasil.com.br/t/sms-brasil-ups-add-on-para-nobreaks/2689/18) so I thought it would be better to post here. 

So I have HomeAssistant OS 7.2 running on VMWare. I connected the USB to the virtual machine and checked, it seems to be on the `/dev/ttyUSB0` path but when I try to start the addon it gives me an error:

```
Can't start addon_e52e67e4_hassio_sms_ups: 500 Server Error for http+docker://localhost/v1.41/containers/249862b051c798cbf443b30c2b40b3513603414592b9901da4f699fdef92c71c/start: Internal Server Error ("error while creating mount source path '/sys/firmware/devicetree/base': mkdir /sys/firmware/devicetree: operation not permitted")
``` 

![image](https://user-images.githubusercontent.com/2729095/152624127-111885e5-9f2c-4db5-b05d-c7eb68f5ef23.png)

I'm happy to help but I'm not sure how at this point.

**Comment by dmslabsbr** [comment]
Hi @RobertoSchneiders , Could you try to run smsUPS.py outside HA structure and see what is happening?

You could found some instructions here https://github.com/dmslabsbr/smsUps/blob/master/README.md

**Comment by dmslabsbr** [comment]
Descrição: https://forum.homeassistantbrasil.com.br/t/sms-brasil-ups-add-on-para-nobreaks/2689/18

**Comment by dclobato** [comment]
Same problem here... I have the following configuration

```
UPS_NAME: SMS
UPS_ID: '01'
SMSUPS_SERVER: true
SMSUPS_CLIENTE: true
SMSUPS_FULL_POWER: 1400
PORTA: >-
  /dev/serial/by-id/usb-Prolific_Technology_Inc._USB-Serial_Controller_D-if00-port0
allow_shutdown: true
SHUTDOWN_CMD: sudo shutdown -h now, sudo shutdown now, systemctl poweroff, sudo poweroff
USE_SECRETS: true
Long_lived_access_token: >-
  eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiI4MzdhMDYyM2EzYTA0MWQxYjIxMmRkNzdiMTVjZjRkYiIsImlhdCI6MTY0NTY0MzM3OCwiZXhwIjoxOTYxMDAzMzc4fQ.iSkhfJEluKWRr0pwwmDzuKYke_MfG5J7rRPexw6FPzE
DEVELOPERS_MODE: false
MQTT_HOST: 10.0.1.16
MQTT_USER: homeassistant
MQTT_PASS: WRWRpuzUrXcaYJL4
```

The error message is

```
Can't start addon_e52e67e4_hassio_sms_ups: 500 Server Error for http+docker://localhost/v1.41/containers/0cb10f8f9a7b58faca11e99da9cb74a581e1dd8990beaeef9558c7373dbbddf3/start: Internal Server Error ("error while creating mount source path '/sys/firmware/devicetree/base': mkdir /sys/firmware/devicetree: operation not permitted")
```

The supervisor's log has the following relevant lines

```
22-02-23 16:44:10 ERROR (SyncWorker_3) [supervisor.docker] Can't start addon_e52e67e4_hassio_sms_ups: 500 Server Error for http+docker://localhost/v1.41/containers/f52c9c6c702ee1a580b28a3d0dabffa012a43b9aeec1deb4fcc7aaf6eb864164/start: Internal Server Error ("error while creating mount source path '/sys/firmware/devicetree/base': mkdir /sys/firmware/devicetree: operation not permitted")
22-02-23 16:45:17 INFO (SyncWorker_0) [supervisor.docker.interface] Cleaning addon_e52e67e4_hassio_sms_ups application
22-02-23 16:45:17 ERROR (SyncWorker_0) [supervisor.docker] Can't start addon_e52e67e4_hassio_sms_ups: 500 Server Error for http+docker://localhost/v1.41/containers/d3d05b9a17b23551f0cc73bfffa265b3fe3408415bdba34973264849c5dd5a80/start: Internal Server Error ("error while creating mount source path '/sys/firmware/devicetree/base': mkdir /sys/firmware/devicetree: operation not permitted")
22-02-23 16:52:09 WARNING (MainThread) [supervisor.addons.options] Option 'MQTT_HOST' does not exist in the schema for SMS UPS BETA (e52e67e4_hassio_sms_ups)
22-02-23 16:52:09 WARNING (MainThread) [supervisor.addons.options] Option 'MQTT_USER' does not exist in the schema for SMS UPS BETA (e52e67e4_hassio_sms_ups)
22-02-23 16:52:09 WARNING (MainThread) [supervisor.addons.options] Option 'MQTT_PASS' does not exist in the schema for SMS UPS BETA (e52e67e4_hassio_sms_ups)
22-02-23 16:52:09 INFO (SyncWorker_4) [supervisor.docker.interface] Cleaning addon_e52e67e4_hassio_sms_ups application
22-02-23 16:52:09 WARNING (SyncWorker_4) [supervisor.addons.options] Option 'MQTT_HOST' does not exist in the schema for SMS UPS BETA (e52e67e4_hassio_sms_ups)
22-02-23 16:52:09 WARNING (SyncWorker_4) [supervisor.addons.options] Option 'MQTT_USER' does not exist in the schema for SMS UPS BETA (e52e67e4_hassio_sms_ups)
22-02-23 16:52:09 WARNING (SyncWorker_4) [supervisor.addons.options] Option 'MQTT_PASS' does not exist in the schema for SMS UPS BETA (e52e67e4_hassio_sms_ups)
22-02-23 16:52:09 ERROR (SyncWorker_4) [supervisor.docker] Can't start addon_e52e67e4_hassio_sms_ups: 500 Server Error for http+docker://localhost/v1.41/containers/b20d68ccb4ce096825cf38b6bb2a58355042c90df5c640d9a59fbd78a4cc4261/start: Internal Server Error ("error while creating mount source path '/sys/firmware/devicetree/base': mkdir /sys/firmware/devicetree: operation not permitted")
22-02-23 16:52:46 WARNING (MainThread) [supervisor.addons.options] Option 'MQTT_HOST' does not exist in the schema for SMS UPS BETA (e52e67e4_hassio_sms_ups)
22-02-23 16:52:46 WARNING (MainThread) [supervisor.addons.options] Option 'MQTT_USER' does not exist in the schema for SMS UPS BETA (e52e67e4_hassio_sms_ups)
22-02-23 16:52:46 WARNING (MainThread) [supervisor.addons.options] Option 'MQTT_PASS' does not exist in the schema for SMS UPS BETA (e52e67e4_hassio_sms_ups)
22-02-23 16:52:46 INFO (SyncWorker_1) [supervisor.docker.interface] Cleaning addon_e52e67e4_hassio_sms_ups application
22-02-23 16:52:46 WARNING (SyncWorker_1) [supervisor.addons.options] Option 'MQTT_HOST' does not exist in the schema for SMS UPS BETA (e52e67e4_hassio_sms_ups)
22-02-23 16:52:46 WARNING (SyncWorker_1) [supervisor.addons.options] Option 'MQTT_USER' does not exist in the schema for SMS UPS BETA (e52e67e4_hassio_sms_ups)
22-02-23 16:52:46 WARNING (SyncWorker_1) [supervisor.addons.options] Option 'MQTT_PASS' does not exist in the schema for SMS UPS BETA (e52e67e4_hassio_sms_ups)
22-02-23 16:52:46 ERROR (SyncWorker_1) [supervisor.docker] Can't start addon_e52e67e4_hassio_sms_ups: 500 Server Error for http+docker://localhost/v1.41/containers/0cb10f8f9a7b58faca11e99da9cb74a581e1dd8990beaeef9558c7373dbbddf3/start: Internal Server Error ("error while creating mount source path '/sys/firmware/devicetree/base': mkdir /sys/firmware/devicetree: operation not permitted")
```

Last, but not least, where can I find the ``secrets.ini`` file to fill in the MQTT server parameters?  I can not set them on the configuration screen, right?

**Comment by dmslabsbr** [comment]
Hi, I just push a new version, could you try?

It have more debug info.

**Comment by dclobato** [comment]
> Hi, I just push a new version, could you try?
> 
> It have more debug info.

Updated addon.  No good :(

```
22-02-24 01:23:51 INFO (MainThread) [supervisor.backups.manager] Creating new partial backup with slug 6ed48f1c
22-02-24 01:23:51 INFO (MainThread) [supervisor.backups.manager] Backing up 6ed48f1c store Add-ons
22-02-24 01:24:01 INFO (SyncWorker_0) [supervisor.docker.addon] Export image e52e67e4/amd64-addon-hassio_sms_ups to /data/tmp/tmpjyt5flgh/image.tar
22-02-24 01:24:03 INFO (SyncWorker_0) [supervisor.docker.addon] Export image e52e67e4/amd64-addon-hassio_sms_ups done
22-02-24 01:24:03 INFO (MainThread) [supervisor.addons.addon] Building backup for add-on e52e67e4_hassio_sms_ups
22-02-24 01:25:47 INFO (MainThread) [supervisor.addons.addon] Finish backup for addon e52e67e4_hassio_sms_ups
22-02-24 01:25:48 INFO (MainThread) [supervisor.backups.manager] Creating partial backup with slug 6ed48f1c completed
22-02-24 01:25:48 INFO (SyncWorker_0) [supervisor.docker.interface] Updating image e52e67e4/amd64-addon-hassio_sms_ups:0.40 to e52e67e4/amd64-addon-hassio_sms_ups:0.40aa1
22-02-24 01:25:48 INFO (SyncWorker_0) [supervisor.docker.addon] Starting build for e52e67e4/amd64-addon-hassio_sms_ups:0.40aa1
22-02-24 01:26:06 INFO (SyncWorker_0) [supervisor.docker.addon] Build e52e67e4/amd64-addon-hassio_sms_ups:0.40aa1 done
22-02-24 01:26:06 INFO (SyncWorker_0) [supervisor.docker.interface] Cleaning addon_e52e67e4_hassio_sms_ups application
22-02-24 01:26:06 INFO (MainThread) [supervisor.addons] Add-on 'e52e67e4_hassio_sms_ups' successfully updated
22-02-24 01:26:06 INFO (SyncWorker_1) [supervisor.docker.interface] Cleanup images: ['e52e67e4/amd64-addon-hassio_sms_ups:0.40']
22-02-24 01:26:13 WARNING (MainThread) [supervisor.addons.options] Option 'MQTT_HOST' does not exist in the schema for SMS UPS BETA (e52e67e4_hassio_sms_ups)
22-02-24 01:26:13 WARNING (MainThread) [supervisor.addons.options] Option 'MQTT_USER' does not exist in the schema for SMS UPS BETA (e52e67e4_hassio_sms_ups)
22-02-24 01:26:13 WARNING (MainThread) [supervisor.addons.options] Option 'MQTT_PASS' does not exist in the schema for SMS UPS BETA (e52e67e4_hassio_sms_ups)
22-02-24 01:26:35 ERROR (SyncWorker_4) [supervisor.docker] Can't start addon_e52e67e4_hassio_sms_ups: 500 Server Error for http+docker://localhost/v1.41/containers/73fc3ee06dcddde31b68e5f5f0a14a11c6f580dc21b6b0a4e6abdf4f02af5c68/start: Internal Server Error ("error while creating mount source path '/sys/firmware/devicetree/base': mkdir /sys/firmware/devicetree: operation not permitted")
22-02-24 01:27:07 INFO (SyncWorker_2) [supervisor.docker.interface] Cleaning addon_e52e67e4_hassio_sms_ups application
22-02-24 01:27:07 INFO (SyncWorker_2) [supervisor.docker.interface] Removing image e52e67e4/amd64-addon-hassio_sms_ups with latest and 0.40aa1
22-02-24 01:27:07 INFO (SyncWorker_3) [supervisor.docker.addon] Starting build for e52e67e4/amd64-addon-hassio_sms_ups:0.40aa1
22-02-24 01:27:47 INFO (SyncWorker_3) [supervisor.docker.addon] Build e52e67e4/amd64-addon-hassio_sms_ups:0.40aa1 done
22-02-24 01:27:47 INFO (MainThread) [supervisor.addons] Add-on 'e52e67e4_hassio_sms_ups' successfully rebuilt
22-02-24 01:27:58 ERROR (SyncWorker_3) [supervisor.docker] Can't start addon_e52e67e4_hassio_sms_ups: 500 Server Error for http+docker://localhost/v1.41/containers/a874ed2018f6c04554ecfd2704eb229b0f43ba2244ea5324624a89023b19bf50/start: Internal Server Error ("error while creating mount source path '/sys/firmware/devicetree/base': mkdir /sys/firmware/devicetree: operation not permitted")
```

**Comment by dclobato** [comment]
Updated to 0.41b.  Same error...

```
22-02-24 16:40:53 INFO (MainThread) [supervisor.homeassistant.api] Updated Home Assistant API token
22-02-24 16:49:56 INFO (MainThread) [supervisor.backups.manager] Creating new partial backup with slug 18e7c4ba
22-02-24 16:49:56 INFO (MainThread) [supervisor.backups.manager] Backing up 18e7c4ba store Add-ons
22-02-24 16:50:02 INFO (SyncWorker_4) [supervisor.docker.addon] Export image e52e67e4/amd64-addon-hassio_sms_ups to /data/tmp/tmpf8xb5c1b/image.tar
22-02-24 16:50:04 INFO (SyncWorker_4) [supervisor.docker.addon] Export image e52e67e4/amd64-addon-hassio_sms_ups done
22-02-24 16:50:04 INFO (MainThread) [supervisor.addons.addon] Building backup for add-on e52e67e4_hassio_sms_ups
22-02-24 16:51:48 INFO (MainThread) [supervisor.addons.addon] Finish backup for addon e52e67e4_hassio_sms_ups
22-02-24 16:51:49 INFO (MainThread) [supervisor.backups.manager] Creating partial backup with slug 18e7c4ba completed
22-02-24 16:51:49 INFO (SyncWorker_3) [supervisor.docker.interface] Updating image e52e67e4/amd64-addon-hassio_sms_ups:0.40aa1 to e52e67e4/amd64-addon-hassio_sms_ups:0.41b
22-02-24 16:51:49 INFO (SyncWorker_3) [supervisor.docker.addon] Starting build for e52e67e4/amd64-addon-hassio_sms_ups:0.41b
22-02-24 16:52:06 INFO (SyncWorker_3) [supervisor.docker.addon] Build e52e67e4/amd64-addon-hassio_sms_ups:0.41b done
22-02-24 16:52:06 INFO (SyncWorker_3) [supervisor.docker.interface] Cleaning addon_e52e67e4_hassio_sms_ups application
22-02-24 16:52:06 INFO (MainThread) [supervisor.addons] Add-on 'e52e67e4_hassio_sms_ups' successfully updated
22-02-24 16:52:06 INFO (SyncWorker_1) [supervisor.docker.interface] Cleanup images: ['e52e67e4/amd64-addon-hassio_sms_ups:0.40aa1']
22-02-24 16:52:56 INFO (SyncWorker_5) [supervisor.docker.interface] Removing image e52e67e4/amd64-addon-hassio_sms_ups with latest and 0.41b
22-02-24 16:52:56 INFO (SyncWorker_3) [supervisor.docker.addon] Starting build for e52e67e4/amd64-addon-hassio_sms_ups:0.41b
22-02-24 16:53:35 INFO (SyncWorker_3) [supervisor.docker.addon] Build e52e67e4/amd64-addon-hassio_sms_ups:0.41b done
22-02-24 16:53:35 INFO (MainThread) [supervisor.addons] Add-on 'e52e67e4_hassio_sms_ups' successfully rebuilt
22-02-24 16:55:54 ERROR (SyncWorker_0) [supervisor.docker] Can't start addon_e52e67e4_hassio_sms_ups: 500 Server Error for http+docker://localhost/v1.41/containers/bccc857d7496e8e2e9f85e8a305d2943798731867da35930fae245259c5bcb16/start: Internal Server Error ("error while creating mount source path '/sys/firmware/devicetree/base': mkdir /sys/firmware/devicetree: operation not permitted")
22-02-24 16:56:22 INFO (SyncWorker_5) [supervisor.docker.interface] Cleaning addon_e52e67e4_hassio_sms_ups application
22-02-24 16:56:22 ERROR (SyncWorker_5) [supervisor.docker] Can't start addon_e52e67e4_hassio_sms_ups: 500 Server Error for http+docker://localhost/v1.41/containers/be09515b805191ad7b4e619821f5671ce8082018c6830c9eb5f81c6117ee385b/start: Internal Server Error ("error while creating mount source path '/sys/firmware/devicetree/base': mkdir /sys/firmware/devicetree: operation not permitted")
```

**Comment by dmslabsbr** [comment]
@dclobato , I guess v42 could solve it!

Could you check again?

**Comment by dmslabsbr** [comment]
@RobertoSchneiders , I guess v42 could solve it!

Could you check it?

**Comment by dclobato** [comment]
> @dclobato , I guess v42 could solve it!
> 
> Could you check again?

Sort of... :)

The ``devicetree`` error has gone for good, but where can I set MQTT user, password and server?  I understand that I can set them on ``secrets.ini``, but where should I store this file?

```
templates:
total 16
drwxr-xr-x    2 root     root          4096 Feb 25 15:17 .
drwxr-xr-x    4 root     root          4096 Feb 25 15:21 ..
-rw-r--r--    1 root     root           718 Feb 25 15:17 base.html
-rw-r--r--    1 root     root           221 Feb 25 15:17 index.html
/data
total 16
drwxr-xr-x    2 root     root          4096 Feb 25 15:21 .
drwxr-xr-x    4 root     root          4096 Feb 25 15:21 ..
-rw-r--r--    1 root     root           718 Feb 25 15:21 base.html
-rw-r--r--    1 root     root           221 Feb 25 15:21 index.html
PATH: 
MQTT_HOST -
MQTT_PASS -
Serial ports
/data
total 0
drwxr-xr-x    2 root     root            80 Feb 23 16:38 .
drwxr-xr-x    4 root     root            80 Feb 23 16:38 ..
lrwxrwxrwx    1 root     root            13 Feb 23 16:38 usb-1a86_USB_Serial-if00-port0 -> ../../ttyUSB0
lrwxrwxrwx    1 root     root            13 Feb 23 16:38 usb-Prolific_Technology_Inc._USB-Serial_Controller_D-if00-port0 -> ../../ttyUSB1
Bus 001 Device 001: ID 1d6b:0002
Bus 001 Device 004: ID 31b2:8001
Bus 001 Device 003: ID 067b:2303
Bus 001 Device 002: ID 1a86:7523
Bus 002 Device 001: ID 1d6b:0003
Bus 001 Device 005: ID 8087:0a2a
[15:21:13] INFO: secrets.ini exists!
[15:21:13] INFO: /data/secrets.ini exists!
SMS BRASIL UPS - dmslabs
********** dmslabs smsUPS v.0.41
Starting up... 2022-02-25 15:21:13
2022-02-25 15:21:13,762 - root - DEBUG - ********** dmslabs smsUPS v.0.41
2022-02-25 15:21:13,762 - root - DEBUG - Starting up...
Path: /data
2022-02-25 15:21:13,762 - root - DEBUG - Path: /data
hassio_token:19029c2462e6f48c2258b2d43d55091b76a60bda111cc3ce8c5e4f0722b4b4fb4de82aa77395f8edd71524d5b281931f794d72454392057b
2022-02-25 15:21:13,762 - root - DEBUG - Running inside HASSIO True
2022-02-25 15:21:13,762 - root - DEBUG - env1:/dev/ttyUSB1
Getting Config Parser.
2022-02-25 15:21:13,762 - root - DEBUG - Existe secrets.ini
Existe secrets.ini
Reading secrets.ini
2022-02-25 15:21:13,764 - root - WARNING - Reading secrets.ini
2022-02-25 15:21:13,765 - root - DEBUG - Config: SMSUPS_FULL_POWER use default: 1450
Getting Config Parser.
2022-02-25 15:21:13,765 - root - DEBUG - Existe secrets.ini
Existe secrets.ini
2022-02-25 15:21:13,767 - root - DEBUG - Config: ALLOW_SHUTDOWN use default: True
2022-02-25 15:21:13,767 - root - DEBUG - Config: DEVELOPERS_MODE use default: False
False
2022-02-25 15:21:13,768 - root - DEBUG - PORTA secrets: /dev/tty.usbserial-1440/dev/tty.usbserial-1470/dev/ttyUSB0/devx/ttyUSB0
 NORMAL MODE 
 IN HASS.IO 
2022-02-25 15:21:13,768 - root - DEBUG - Loading env data....
false
Getting Config Parser.
2022-02-25 15:21:13,768 - root - DEBUG - Existe secrets.ini
Existe secrets.ini
2022-02-25 15:21:13,770 - root - DEBUG - Env data loaded.
2022-02-25 15:21:13,770 - root - DEBUG - PORTA substitui: /dev/ttyUSB1
Getting Config Parser.
2022-02-25 15:21:13,770 - root - DEBUG - Existe secrets.ini
Existe secrets.ini
Reading secrets.ini
2022-02-25 15:21:13,772 - root - WARNING - Reading secrets.ini
2022-02-25 15:21:13,773 - root - DEBUG - Config: SMSUPS_FULL_POWER use default: 1400
Getting Config Parser.
2022-02-25 15:21:13,773 - root - DEBUG - Existe secrets.ini
Existe secrets.ini
2022-02-25 15:21:13,775 - root - DEBUG - Config: ALLOW_SHUTDOWN use default: True
2022-02-25 15:21:13,775 - root - DEBUG - Config: DEVELOPERS_MODE use default: false
false
2022-02-25 15:21:13,775 - root - DEBUG - SMSUPS_SERVER: True
2022-02-25 15:21:13,775 - root - DEBUG - SMSUPS_CLIENTE: True
2022-02-25 15:21:13,775 - root - DEBUG - MQTT_HOST: 192.168.50.20
2022-02-25 15:21:13,776 - root - DEBUG - IP: 10.0.1.15
SMSUPS_SERVER: True
SMSUPS_CLIENTE: True
IP: 10.0.1.15
UPS_ID: 01
UPS_NAME: SMS
UPS_NAME_ID: ups_SMS_01
ALLOW_SHUTDOWN: True
Getting OS Data
2022-02-25 15:21:13,776 - root - INFO - os.name: posix
Can't get OS Data
info / Error! Code: OSError, Message, unable to determine login name
sh: /ls: not found
2022-02-25 15:21:13,776 - root - DEBUG - info / Error! Code: OSError, Message, unable to determine login name
info / Error! Code: OSError, Message, unable to determine login name
2022-02-25 15:21:13,776 - root - DEBUG - info / Error! Code: OSError, Message, unable to determine login name
info / Error! Code: OSError, Message, unable to determine login name
2022-02-25 15:21:13,776 - root - DEBUG - info / Error! Code: OSError, Message, unable to determine login name
 /dev/ttyUSB0n/dev/ttyUSB0o/dev/ttyUSB0t/dev/ttyUSB0 /dev/ttyUSB0e/dev/ttyUSB0x/dev/ttyUSB0i/dev/ttyUSB0s/dev/ttyUSB0t
2022-02-25 15:21:13,778 - root - INFO - Starting MQTT 192.168.50.20
2022-02-25 15:21:13,779 - root - DEBUG - mqttStart MQTT_PASSWORD: your_mqtt_file
MQTT Start / Error! Code: timeout, Message, timed out
2022-02-25 15:21:18,785 - root - WARNING - MQTT Start / Error! Code: timeout, Message, timed out
```

**Comment by dmslabsbr** [comment]
@dclobato , it should get the mqqt server, user and password from HASS.IO.    Do you use an external MQQT server?   I going to check, I guess it getting a wrong login data.

**Comment by dclobato** [comment]
> @dclobato , it should get the mqqt server, user and password from HASS.IO. Do you use an external MQQT server? I going to check, I guess it getting a wrong login data.

Yes, I do

Where can I set/edit the ``secrets.ini`` on the HASSI.IO?

**Comment by RobertoSchneiders** [comment]
@dmslabsbr sorry for the late response. This issue seems to be fixed on v0.42, thank you. 

I don't see any sensors/entities after the addon starts, I'm probably facing a different issue now. any ideas? (UPS is an SMS Power Sinus NG 3200 VA)

```
Serial ports
total 0
drwxr-xr-x    2 root     root            60 Mar  3 22:13 .
drwxr-xr-x    4 root     root            80 Mar  3 22:13 ..
lrwxrwxrwx    1 root     root            13 Mar  3 22:13 usb-Prolific_Technology_Inc._USB-Serial_Controller_D-if00-port0 -> ../../ttyUSB0
Bus 003 Device 001: ID 1d6b:0002
Bus 002 Device 002: ID 0e0f:0003
Bus 001 Device 001: ID 1d6b:0002
Bus 004 Device 001: ID 1d6b:0003
Bus 002 Device 003: ID 0e0f:0002
Bus 002 Device 001: ID 1d6b:0001
Bus 003 Device 002: ID 067b:2303
[22:14:12] INFO: secrets.ini exists!
[22:14:12] INFO: /data/secrets.ini exists!
SMS BRASIL UPS - dmslabs
********** dmslabs smsUPS v.0.41
Starting up... 2022-03-03 22:14:13
2022-03-03 22:14:13,120 - root - DEBUG - ********** dmslabs smsUPS v.0.41
2022-03-03 22:14:13,120 - root - DEBUG - Starting up...
Path: /data
2022-03-03 22:14:13,120 - root - DEBUG - Path: /data
hassio_token:some_hash
2022-03-03 22:14:13,120 - root - DEBUG - Running inside HASSIO True
2022-03-03 22:14:13,120 - root - DEBUG - env1:/dev/ttyUSB0
Getting Config Parser.
2022-03-03 22:14:13,120 - root - DEBUG - Existe secrets.ini
Existe secrets.ini
Reading secrets.ini
2022-03-03 22:14:13,121 - root - WARNING - Reading secrets.ini
2022-03-03 22:14:13,121 - root - DEBUG - Config: SMSUPS_FULL_POWER use default: 1450
Getting Config Parser.
2022-03-03 22:14:13,121 - root - DEBUG - Existe secrets.ini
Existe secrets.ini
2022-03-03 22:14:13,121 - root - DEBUG - Config: ALLOW_SHUTDOWN use default: True
2022-03-03 22:14:13,121 - root - DEBUG - Config: DEVELOPERS_MODE use default: False
False
2022-03-03 22:14:13,121 - root - DEBUG - PORTA secrets: /dev/tty.usbserial-1440/dev/tty.usbserial-1470/dev/ttyUSB0/devx/ttyUSB0
 NORMAL MODE 
 IN HASS.IO 
2022-03-03 22:14:13,121 - root - DEBUG - Loading env data....
false
Getting Config Parser.
2022-03-03 22:14:13,121 - root - DEBUG - Existe secrets.ini
Existe secrets.ini
2022-03-03 22:14:13,122 - root - DEBUG - Env data loaded.
2022-03-03 22:14:13,122 - root - DEBUG - PORTA substitui: /dev/ttyUSB0
2022-03-03 22:14:13,122 - root - DEBUG - SMSUPS_SERVER: True
2022-03-03 22:14:13,122 - root - DEBUG - SMSUPS_CLIENTE: False
2022-03-03 22:14:13,122 - root - DEBUG - MQTT_HOST: core-mosquitto
2022-03-03 22:14:13,122 - root - DEBUG - IP: 192.168.0.147
SMSUPS_SERVER: True
SMSUPS_CLIENTE: False
IP: 192.168.0.147
UPS_ID: 01
UPS_NAME: SMS
UPS_NAME_ID: ups_SMS_01
ALLOW_SHUTDOWN: False
Getting OS Data
2022-03-03 22:14:13,122 - root - INFO - os.name: posix
Can't get OS Data
info / Error! Code: OSError, Message, unable to determine login name
2022-03-03 22:14:13,122 - root - DEBUG - info / Error! Code: OSError, Message, unable to determine login name
info / Error! Code: OSError, Message, unable to determine login name
2022-03-03 22:14:13,122 - root - DEBUG - info / Error! Code: OSError, Message, unable to determine login name
info / Error! Code: OSError, Message, unable to determine login name
2022-03-03 22:14:13,122 - root - DEBUG - info / Error! Code: OSError, Message, unable to determine login name
sh: /ls: not found
 /dev/ttyUSB0n/dev/ttyUSB0o/dev/ttyUSB0t/dev/ttyUSB0 /dev/ttyUSB0e/dev/ttyUSB0x/dev/ttyUSB0i/dev/ttyUSB0s/dev/ttyUSB0t
2022-03-03 22:14:13,123 - root - INFO - Starting MQTT core-mosquitto
2022-03-03 22:14:13,123 - root - DEBUG - mqttStart MQTT_PASSWORD: 
MQTT connected with result code 0
2022-03-03 22:14:13,124 - root - DEBUG - MQTT connected with result code 0
Connected to core-mosquitto
2022-03-03 22:14:13,124 - root - INFO - Subscribe Topic: home/ups/cmd
Subscribe Topic: home/ups/cmd
Porta: /dev/ttyUSB0 - True
2022-03-03 22:14:14,128 - root - INFO - Port /dev/ttyUSB0 - is open: True
portstr: /dev/ttyUSB0
2022-03-03 22:14:15,930 - root - DEBUG - cmd:Name / str: 49 ff ff ff ff bb 0d / Q: False
2022-03-03 22:14:26,643 - root - DEBUG - response: b'3a506f7765722053696e757320312e32d60d'
2022-03-03 22:14:26,643 - root - DEBUG - cmd:Info / str: 46 ff ff ff ff be 0d / Q: False
2022-03-03 22:14:37,355 - root - DEBUG - response: b'3b323230202020203030302032343630350d'
2022-03-03 22:14:37,355 - root - DEBUG - UPS Info: SMS / no info
2022-03-03 22:14:37,355 - root - DEBUG - UPS Info: SMS / no info
2022-03-03 22:14:37,355 - root - DEBUG - Sensor_dic: 0
2022-03-03 22:14:37,358 - root - DEBUG - Hass Sended
Web server outorder
Loop start!
Dados UPS Publicados...2022-03-03 22:14:48.063317
Dados UPS Publicados...2022-03-03 22:15:01.779816
Dados UPS Publicados...2022-03-03 22:15:15.499186
Dados UPS Publicados...2022-03-03 22:15:42.961890
Dados UPS Publicados...2022-03-03 22:15:56.682016
Dados UPS Publicados...2022-03-03 22:16:24.119400
Dados UPS Publicados...2022-03-03 22:16:37.839548
Dados UPS Publicados...2022-03-03 22:16:51.559872
2022-03-03 22:16:51,560 - root - DEBUG - Sensor_dic: 3
2022-03-03 22:16:51,562 - root - DEBUG - Hass Sended
```
